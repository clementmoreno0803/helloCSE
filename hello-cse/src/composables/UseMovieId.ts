import { useRoute } from "vue-router";

export const useMovieId = () => {
  const route = useRoute();
  const movieId = String(route.params.id);
  return movieId;
};
