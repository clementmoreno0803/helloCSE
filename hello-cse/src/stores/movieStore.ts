import { movieDetail } from "@/models/movieDetailInterface";
import { topMovieInterface } from "@/models/topMovieInterface";
import { upComingMoviesInterface } from "@/models/upCommingMovieInterface";
import { defineStore } from "pinia";
import { computed, ref } from "vue";
import { currentMovieInterface } from "@/models/currentMovieInterface";

export const useMovieStore = defineStore("movieStore", () => {
  const topMovies = ref<topMovieInterface[]>([]);
  const currentMovies = ref<currentMovieInterface[]>([]);
  const upComingMovies = ref<upComingMoviesInterface[]>([]);
  const movieDetails = ref<movieDetail>();
  const inputFilter = ref<string>("");
  const isLoaded = ref<boolean>(false);

  //getter

  const getFilteredMovies = computed(() => {
    return upComingMovies.value.filter((movie) => movie.title.toLowerCase().includes(inputFilter.value.toLowerCase()));
  });

  const getTopFiveMovies = computed(() => {
    const moviesSorted = topMovies.value.sort((a, b) => b.vote_average - a.vote_average);
    return moviesSorted.slice(0, 5);
  });

  const getHeightCurrentMovies = computed(() => {
    return currentMovies.value.slice(0, 8);
  });

  const listTop4Actors = computed(() => {
    const castingSorted = movieDetails.value?.cast.sort((a, b) => b.popularity - a.popularity);
    return castingSorted?.slice(0, 4);
  });

  //setter
  const setTopMovies = (newMovies: topMovieInterface[]) => {
    topMovies.value = newMovies;
  };
  const setCurrentMovies = (newCurrentMovie: currentMovieInterface[]) => {
    currentMovies.value = newCurrentMovie;
  };
  const setUpComingMovies = (newMovies: upComingMoviesInterface[]) => {
    upComingMovies.value = newMovies;
    isLoaded.value = true;
  };
  const setMovieFilterByName = (input: string) => {
    inputFilter.value = input;
  };
  const setMovieDetails = (details: movieDetail | undefined) => {
    if (details) {
      return (movieDetails.value = details);
    }
  };

  return {
    topMovies,
    upComingMovies,
    inputFilter,
    getFilteredMovies,
    getTopFiveMovies,
    getHeightCurrentMovies,
    movieDetails,
    listTop4Actors,
    isLoaded,
    setTopMovies,
    setCurrentMovies,
    setUpComingMovies,
    setMovieFilterByName,
    setMovieDetails
  };
});
