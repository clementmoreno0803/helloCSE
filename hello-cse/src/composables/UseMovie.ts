import { useMovieService } from "@/services/UseMovieService";
import { useMovieStore } from "@/stores/movieStore";

export const useMovie = () => {
  const { topMovies } = useMovieService();
  const { setTopMovies } = useMovieStore();

  const getTopMovies = async () => {
    const topMoviesArray = await topMovies();
    console.log(topMoviesArray, "topMovies");
    setTopMovies(topMoviesArray);
  };

  return { getTopMovies };
};
