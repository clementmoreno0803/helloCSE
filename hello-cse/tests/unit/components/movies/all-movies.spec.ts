import { describe, it, expect, vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import { useMovie } from "@/composables/UseMovie"; // Simuler la fonction de composable
import { createRouter, createWebHistory } from 'vue-router';
import { createTestingPinia } from '@pinia/testing'
import AllMovies from "@/components/movies/all-movies.vue";
import { mockMovies } from "../../fixtures/mockMovies.mock";
import { useMovieStore } from "@/stores/movieStore";
import { storeToRefs } from "pinia";

// Configuration du routeur
const router = createRouter({
  history: createWebHistory(),
  routes: [
    {
      path: '/movie-detail/:id',
      name: 'MovieDetail',
      component: { template: '<div>Movie Detail</div>' },
    },
  ],
});

const generateWrapper = () => {
  return shallowMount(AllMovies, {
    global: {
      plugins: [router, createTestingPinia({
        initialState: {
          movieStore: {
            upComingMovies: [],
            getFilteredMovies: [] ,
            isLoaded: false ,
          }
        },
        createSpy: vi.fn,
      })],
    },
  })
}

describe('getAllComments', () => {
  it('should render correctly while loading', () => {
    const wrapper = generateWrapper()

    expect(wrapper.find('.skeleton-title').exists()).toBe(true);
    expect(wrapper.find('.all-movies').exists()).toBe(false);
  });

  it('should load more movies on scroll', async () => {
    const wrapper = generateWrapper();
    const movieStore = useMovieStore()
    movieStore.isLoaded = true;
    movieStore.upComingMovies = mockMovies;
    await wrapper.vm.$nextTick()

    await wrapper.vm.load({ done: vi.fn() });

    expect(movieStore.upComingMovies.length).toBe(mockMovies.length); // Vérifie que les films sont chargés
  });

  it('should navigate to movie detail page', async () => {
    const wrapper = generateWrapper();
    const movieStore = useMovieStore();

    movieStore.isLoaded = true;
    movieStore.upComingMovies = mockMovies;
    await wrapper.vm.$nextTick();

    const movieCard = wrapper.findAll('.all-movies__card-container')[0];
    await movieCard.trigger('click');

    console.log(wrapper.vm.$route.path);

    expect(wrapper.vm.goToDetail).toHaveBeenCalled();
  })
});
