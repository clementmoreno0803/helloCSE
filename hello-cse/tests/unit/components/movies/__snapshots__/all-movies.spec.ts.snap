// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AllMovies.vue > should render correctly 1`] = `
<div
  data-v-46e237f8=""
>
  <div
    data-v-46e237f8=""
  >
    <v-skeleton-loader
      class="skeleton-title"
      color="rgb(28,28,28)"
      data-v-46e237f8=""
      elevation="3"
      type="paragraph"
    />
    <v-skeleton-loader
      class="top-movie-card__carousel-container"
      color="rgb(28,28,28)"
      data-v-46e237f8=""
      elevation="3"
      type="card"
    />
  </div>
</div>
`;
